### ISSUE TITLE
False positives for Squiz.PHP.CommentedOutCode.Found in 3.2.3

### ISSUE THREAD

**Issue by mundschenk-at** [squizlabs/PHP_CodeSniffer#1910]
When updating to 3.2.3, I noticed several new warnings for `Squiz.PHP.CommentedOutCode.Found`. Looking at the code in question, those were all triggered by lines like this:

```php
return false; // @codeCoverageIgnore
```

While `@` could be construed as the error suppression operator, annotations like this should be ignored by the sniff, as the were in previous versions up to 3.2.2.

**Comment by jrfnl** [comment]
Related to #1839 

This is caused by commit 2db7ad4425eac71115e420ff8607c0e3f17c8217 which is an efficiency fix.

Previously, that comment would be re-tokenized as the below - which would result in a 34% commented out code percentage, which doesn't meet the `35` threshold.:
```
0 :: L001 :: C1 :: T_OPEN_TAG :: (6) :: <?php
1 :: L001 :: C7 :: T_WHITESPACE :: (1) ::
2 :: L001 :: C8 :: T_ASPERAND :: (1) :: @
3 :: L001 :: C9 :: T_STRING :: (18) :: codeCoverageIgnore
4 :: L001 :: C27 :: T_WHITESPACE :: (1) ::
5 :: L001 :: C28 :: T_CLOSE_TAG :: (2) :: ?>
```

Now it is re-tokenized as the below - note the missing additional whitespace token at position 1 -. This now results in a 50% commented out code percentage, causing it to be reported.
```
0 :: L001 :: C1 :: T_OPEN_TAG :: (6) :: <?php
1 :: L001 :: C7 :: T_ASPERAND :: (1) :: @
2 :: L001 :: C8 :: T_STRING :: (18) :: codeCoverageIgnore
3 :: L001 :: C26 :: T_WHITESPACE :: (1) ::
4 :: L001 :: C27 :: T_CLOSE_TAG :: (2) :: ?>
```

To be fair, in most cases, this is a bug _fix_. The sniff was underreporting commented out code.

For your particular example, however, I totally understand your annoyance at the fact that - now the bug is fixed - you start seeing these messages.

There are several ways I can think of to fix this, but I am not sure what would be the preferred way in this case, so I'd like to ask @gsherwood to please advise and I'd be happy to implement a fix for this.

Suggestions:
* Add a public property to allow users to pass an array of words/phrases to ignore in comments. These words/phrases would then be filtered out before retokenizing. This would allow for maximum flexibility when dealing with a variety of external tools which each have their own type of ignore comments.
* Hard code an exception for the `codeCoverageIgnore` phrase.
* Hard code an exception for any comment which only contains a `@` + one word.

**Comment by mundschenk-at** [comment]
Thanks for the quick analysis, @jrfnl! I think either the first or third suggestion would be preferable. [PHPUnit has three code coverage annotations](https://phpunit.de/manual/current/en/code-coverage-analysis.html) and there are probably scores of other tools that work in a similar way.

**Comment by gsherwood** [comment]
Option 3 would be my preference.

**Comment by jrfnl** [comment]
@gsherwood Just to make sure: Option 3 would not solve the "false positives" as reported in #1912. (Option 5 would for the most part).

**Comment by gsherwood** [comment]
> @gsherwood Just to make sure: Option 3 would not solve the "false positives" as reported in #1912. (Option 5 would for the most part).

Sorry, was just getting around to commenting again once I closed that in favour of this one.

After seeing those examples, I think the 2-or-less non-whitespace token idea might have to be the way to go, along with option 3 for any code using @-style annotations.

**Comment by jrfnl** [comment]
See PR #1914
